powerscale/helper/user_mapping_rules_helper.go

```diff
@@ -0,0 +1,433 @@
+/*
+Copyright (c) 2024 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"context"
+	powerscale "dell/powerscale-go-client"
+	"fmt"
+	"terraform-provider-powerscale/client"
+	"terraform-provider-powerscale/powerscale/constants"
+	"terraform-provider-powerscale/powerscale/models"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// UpdateUserMappingRulesState updates resource state.
+func UpdateUserMappingRulesState(ctx context.Context, rulesState *models.UserMappingRulesResourceModel, rulesResponse *powerscale.V1MappingUsersRulesRules) (diags diag.Diagnostics) {
```

Comment:
Is it possible to use `CopyFields` or whatever common method to update resource model, rather than updating fields one by one?

Reply:
@baoy1  CopyFields need to use its tdsk tag, but current resource elements' type are types.Object or types.List, they has no tdsk tag, so we need to use this way to parse their values currently.
![image](https://github.com/dell/terraform-provider-powerscale/assets/88763781/9d5fde79-5387-49f0-833f-22801126cc71)

Reply:
Yes, if there are many `tfsdk` fields, we can use `CopyFields ` to save the effort. For those fields without `tfsdk`, we can manually set the value. Maybe some enhancement to handle non-tfsdk fields in future...